ga4gh/backend.py

```diff
@@ -17,6 +17,81 @@
 import ga4gh.datamodel.variants as variants
 
 
+def _getReadAlignmentStart(readAlignment):
+    return readAlignment.alignment.position.position
+
+
+def _getReadAlignmentEnd(readAlignment):
+    return _getReadAlignmentStart(readAlignment) + \
+        len(readAlignment.alignedSequence)
+
+
+def _getVariantStart(variant):
+    return variant.start
+
+
+def _getVariantEnd(variant):
+    return variant.end
+
+
+class IntervalIterator(object):
+    """
+    Implements generator logic for types which accept a start/end
+    range to search for the object
+    """
+    def __init__(self, request, iterator, getStart, getEnd):
+        self._request = request
+        self._iterator = iterator
+        self._getStart = getStart
+        self._getEnd = getEnd
+        self._badPageTokenExceptionMessage = (
+            "Inconsistent page token provided")
+
+    def iterate(self, startPosition, equalPositionsToSkip):
```

Comment:
I don't really see much value in splitting up the `_positionIterator` and `_yieldObjects` part here; these are intrinsically linked, and I think it's clearer to see them explicitly laid out together in one method.

Also, I think it might be nice if we implemented the iterator protocol explicitly in this object, so that client code will be able to just call:
```python
return IntervalIterator(request, iterator, _getVariantStart, _getVariantEnd)
```